Sources/Target/Common/ProcessBase.cpp

```diff
@@ -225,6 +225,11 @@ ErrorCode ProcessBase::readMemoryBuffer(Address const &address, size_t length,
     return error;
   }
 
+  SoftwareBreakpointManager *bpm = softwareBreakpointManager();
+  if (bpm) {
+    bpm->insertStashedInsns(address, length, buffer);
+  }
+
```

Comment:
If I understand correctly, you want to make sure that when we perform a memory read, we're not sending the breakpoint instructions we potentially placed. You want to return the actual instruction that is supposed to be there. Is this right?

Reply:
Yep